### ISSUE TITLE
Yarn link doesn't work

### ISSUE THREAD

**Issue by RusPosevkin** [yarnpkg/yarn#1297]
**Do you want to request a _feature_ or report a _bug_?**
Bug.

**What is the current behavior?**

```
foo@bar:~/code/lpserver/frontend/addons/lp-base$ yarn link
yarn link v0.16.1
success Registered "lp-base".
info You can now run `yarn link "lp-base"` in the projects where you want to use this module and it will be used instead.
Done in 0.05s.
foo@bar:~/code/lpserver/frontend/addons/lp-base$ cd ../../apps/partner/
foo@bar:~/code/lpserver/frontend/apps/partner$ yarn link lp-base
yarn link v0.16.1
success Registered "lp-base".
Done in 0.05s.
foo@bar:~/code/lpserver/frontend/apps/partner$ yarn
yarn install v0.16.1
info No lockfile found.
[1/4] Resolving packages...
error Couldn't find package "lp-base" on the "npm" registry.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**If the current behavior is a bug, please provide the steps to reproduce.**
See above.

**What is the expected behavior?**

```
foo@bar:~/code/lpserver/frontend/addons/lp-base$ yarn link
yarn link v0.16.1
success Registered "lp-base".
info You can now run `yarn link "lp-base"` in the projects where you want to use this module and it will be used instead.
Done in 0.05s.
foo@bar:~/code/lpserver/frontend/addons/lp-base$ cd ../../apps/partner/
foo@bar:~/code/lpserver/frontend/apps/partner$ yarn link lp-base
yarn link v0.16.1
success Registered "lp-base".
Done in 0.05s.
foo@bar:~/code/lpserver/frontend/apps/partner$ yarn
yarn install v0.16.1
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 0.46s.
```

**Please mention your node.js, yarn and operating system version.**

```
node 6.4.0
yarn 0.16.1
Ubuntu 16.04 LTS
```

**Comment by RusPosevkin** [comment]
Probably, there is a bug when linked package name include the hyphen. The same problem may be in #1161.

**Comment by ssteinerx** [comment]
Same problem with package with no hyphen in name; it still looks for package in on the "npm" registry.

Dead in the water without this working on my current project.

**Comment by GCorbel** [comment]
I have the same bug. I can't switch to yarn yet.

**Comment by jampy** [comment]
Any news on this?

I think I'm having the same issue (at least the result is the same).

If it helps, some details regarding my situation:

- building inside a Docker container, as "root", within a VM in Windows
- I have three packages locally under `/src-protected/tonidb/packages` named `tonidb`, `tonidb-odm` and `tonidb-server`. The last two depend on `tonidb` and thus are linked using `yarn link tonidb`. **This works** as the integrated tests succeed (`yarn run test`), so linking somehow works.
- The main application under `/src-protected/winfarmer-xt/src` is linked to the three packages just mentioned, but  fails with errors like `Couldn't find package "tonidb" on the "npm" registry.`. The package name in the error message randomly switches between one of the three package names - I guess that's simple because of the parallel install.
- Since the `tonidb*` packages are unreleased to NPM yet, I've added them to the `package.json` files simply like `"tonidb": "*"`.
- Yarn version is v0.17.8.

I have no idea why the link works for the dependencies themselves but not for the main application.

**Comment by jampy** [comment]
Hmmm, anybody found a workaround?

Would like to use Yarn, but this is a major blocker :(

**Comment by cdomigan** [comment]
Yarn link seems completely broken? Would be great to get an update from the devs on this. Is anybody having luck using link at all?

**Comment by RusPosevkin** [comment]
Hey guys, let's try to use symbolic linking
``` bash
  $> ln -nsf ../foo node_modules/foo
  ```

It can be used into package.json:
``` json
"scripts": {
    "link": "(mkdir node_modules || true) && ln -nsf ../foo node_modules/foo && ln -nsf ../bar node_modules/bar",
    "setup": "yarn run link && yarn",
  },
```

Just run setup task:
``` bash
  $> yarn run setup
  ```

**Comment by StarpTech** [comment]
Run into the same issue. After linking and install the package I got:

```
yarn
yarn install v0.23.2
[1/4] Resolving packages...
error Received malformed response from registry for "local-test". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by bestander** [comment]
We are going to revamp link: feature soon, see https://github.com/yarnpkg/rfcs/pull/34

**Comment by voxsim** [comment]
Ok I will wait you guys, if you need some help let me know :)

**Comment by gsklee** [comment]
@RusPosevkin could you confirm that if this issue has been resolved in v0.26 prerelease?

**Comment by fugufish** [comment]
it does not seem to to work on latest prerelease

**Comment by alexre0** [comment]
+1
v0.27.3

**Comment by bestander** [comment]
I could not reproduce on latest yarn@0.27.3.
Closing.

@fugufish and @alexre0, please provide a test script how I can reproduce it not working.
Better open a new issue with the steps.

**Comment by fugufish** [comment]
create a brand new package that is not registered on npm, use `yarn link <package-name>`
create project that references that package and version, use `yarn link <package-name>`
package correctly shows up in node_modules
call `yarn install`
an error will be thrown saying the package does not exist.

**Comment by bestander** [comment]
first time you need to call `yarn link` without package name.

**Comment by fugufish** [comment]
either way it doesn't work.

**Comment by cloakedninjas** [comment]
This is borked for me too - only when my dependency is installed locally can I get things to work, otherwise:

```
var gulp = require('gulp');
```
```
Error: Cannot find module 'gulp'
```

**Comment by bestander** [comment]
I think we are talking about different issues here.

@fugufish, do I understand it right that you `yarn link project-b` and also have a `project-b` in package.json?
I think Yarn may not work with linking and depending at the same time, it's probably a separate bug.
For a workaround you can try `"project-b": "link:./path/to/project-b"`, it will create a symlink when you run `yarn install`.

**Comment by bestander** [comment]
@cloakedninjas, yarn would not run install on linked project, this is expected behavior.
Workspaces feature should address this, it is still experimental

**Comment by cobyism** [comment]
Pretty sure there’s still something messed up with this command. Here’s what I’m running into:

```sh
$ cd ~/code/some-module

$ yarn link

yarn link v0.17.8
success Registered "some-module".
info You can now run `yarn link "some-module"` in the projects where you want to use this module and it will be used instead.
✨  Done in 0.07s.

$ cd ../main-project

$ yarn link "some-module"

yarn link v0.17.8
error No registered module found called "some-module".
info Visit https://yarnpkg.com/en/docs/cli/link for documentation about this command.
```

What am I missing?

![](http://gifs.joelglovier.com/contempt/farley-weekend-update.gif)

**Comment by bestander** [comment]
Can you do `yarn link some-module` without the double quotes?

**Comment by cobyism** [comment]
> Can you do yarn link some-module without the double quotes?

Exactly the same behaviour, regardless of presence/absence or quote type (`'` vs. `"`).

**Comment by bestander** [comment]
Oh wait, your yarn is version 0.17.8, that is very old, can you upgrade to latest?
https://yarnpkg.com/en/docs/install

**Comment by p12y** [comment]
I'm getting the same thing. I've got the latest version of yarn. Create a link in node_modules, navigate to app folder, yarn link [package_name], package disappears from node_modules. Only way I can even fire up a server is by reinstalling the package.

**Comment by lgirma** [comment]
Same issues here.
Using yarn version `1.2.1`, node version `8.4.0`

Running the commands `yarn link` and `yarn link some-module` seems to work fine (at least yarn reports they went well), until I do a build and then get `Cannot find module some-module` error.

**Comment by HazAT** [comment]
I am also running in this issue.
So the behavior I would expect is that `yarn install` skips the modules I already linked before 

e.g.:
```bash
yarn link # in packages/local-test

# somewhere else

yarn link local-test
yarn install 
# no longer tries to install local-test from registry
# even though it's in my package.json as a dependency
```

Currently, I am getting:
```
yarn install v1.3.2
[1/5] Validating package.json...
[2/5] Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/local-test: Not found".
```

**Comment by rightaway** [comment]
Is this working for you in the latest?

**Comment by natew** [comment]
I get this with scoped packages, not sure if that's different.

**Comment by Abdisalan** [comment]
Running into this issue as well, the same way @cobyism  described

**Comment by JimmyCDChen** [comment]
Seeing the same issue with yarn link. In the same boat with @Abdisalan

**Comment by sandalsoft** [comment]
I was getting this error:
`Error: Cannot find module '<package name>'`


What worked for me was:

1. run `yarn link` in the \<package name\> module directory.
2. `yarn build` the \<package name\> module.
3. run `yarn link  <package name>` in the app directory I'm trying to include \<package name\> in.

**Comment by travomic** [comment]
Here's something that threw me off:  The output after linking a module, puts the name of the package in double-quotes (highlighted below) ... when pasting that command, it cant find the linked module unless you remove the double-quotes. e.g: `yarn link @travomic/foo-bar`

<img width="769" alt="screen shot 2018-05-23 at 10 56 11 am" src="https://user-images.githubusercontent.com/37982292/40439579-95cf4c1a-5e78-11e8-9458-fd044f698c02.png">

I poked around in the internals of yarn for a little bit and found `linkRegisteredMessage` gets supplied the package.json config values via the `readRootManifest` method in the config module... But kinda stopped digging.  Perhaps someone with more gumption than I have can make a PR to either handle the link argument with and without the double-quoted value.  Or, simply fix the success message after linking to not show the quotes.

**Comment by dquang-tran** [comment]
Wow this issue has been open for 2 year.

**Comment by wayou** [comment]
met the problem with yarn v1.9.4. :(

**Comment by nemoDreamer** [comment]
@bestander don't you think this issue was prematurely closed? It's obviously not resolved, as there are now multiple duplicate issues complaining about `link` being broken.

Many repro steps outline the exact same behavior:
- `yarn link` in the "@myscope/blahblah" package to be linked
  - ```
    success Registered "@myscope/blahblah".
    info You can now run `yarn link "@myscope/blahblah"` in the projects where you want to use this package and it will be used instead.
    ```
- `ls -l ~/.config/yarn/link/@myscope/`
  - ```
    blahblah -> ../../../../Development/myscope/packages/blahblah
    ```
- `cd` to package you want the link to be used in
- `yarn link "blahblah"` just like the CLI says:
  - ```
    error No registered package found called "@myscope/blahblah".
    ```
- maybe `yarn install`?
  - ```
    The target of linked package "@myscope/blahblah" is missing. Removing link.
    ```
- `ls -l ~/.config/yarn/link/@myscope/`
  - nothing left?!

I have a `"main"` entry, I have some `"bin"` entries, I've tried making it an un-scoped package, etc...

**Comment by bestander** [comment]
@nemoDreamer, thanks for bringing it to my attention.
Looks like the linked module can't be found in some cases and this is common enough to cause so many upvotes.
I would dig into the JS code where "error No registered package found called "@myscope/blahblah"." is generated.
I can open the bug but help from community to find the issue and provide test cases is needed.

**Comment by apfelfabrik** [comment]
I ran into the same issue:

```
➜  dep-a $ yarn link
yarn link v1.15.0-0
success Registered "@company/dep-a".
info You can now run `yarn link "@company/dep-a"` in the projects where you want to use this package and it will be used instead.
✨  Done in 0.34s.

➜  dep-a $ cd ../..

➜  parent $ yarn link "@company/dep-a"
yarn link v1.15.0-0
error No registered package found called "@company/dep-a".
info Visit https://yarnpkg.com/en/docs/cli/link for documentation about this command.
```

The reason this failed for me is that `~/.config` was a symlink itself, so that when creating symlinks in `~/.config/yarn/link` without resolving the real path first, the resulting symlinks for the linked packages are broken.

Quick fix could be to identify symlinks upwards of yarn's link folder. Proper fix should be in this PR: https://github.com/yarnpkg/yarn/pull/7028

**Comment by stephenhurleyjpl** [comment]
@apfelfabrik That PR looks very promising, and is I think exactly what I need! Do you know what's holding up the PR?

**Comment by rodrigoyoshida** [comment]
Same here, struggling for hours to find out a way to make it work.

**Comment by tamzinblake** [comment]
Still encountering the same problem now. Has anyone figured out a fix for this (for scoped packages) yet?

**Comment by stephenhurleyjpl** [comment]
My specific problem was that when I ran `yarn link` in my `dep` repo, it created a symbolic link at `/home/steve/.config/yarn/link/dep` pointing to `/home/steve/projects/dep` (using the absolute path like you see here). However, my `home` folder is a symbolic link within the `/` folder and yarn seems to have trouble with any part of the whole path being a symbolic link, just like apfelfabrik mentioned.

I manually changed the `/home/steve/.config/yarn/link/dep` to instead point to `../../../projects/dep` (`rm dep && ln -s dep ../../../projects/dep`), and this fixed the problem for me.

Still awaiting the PR so that yarn automatically fixes it. The PR is a one-line change, so I'm not sure why it's being held up.

**Comment by skwny** [comment]
> Still encountering the same problem now. Has anyone figured out a fix for this (for scoped packages) yet?

Switched back to npm :)

**Comment by cha0s** [comment]
Please fix.

**Comment by nemoDreamer** [comment]
@apfelfabrik hit the nail on the head!

> The reason this failed for me is that ~/.config was a symlink itself, so that when creating symlinks in ~/.config/yarn/link without resolving the real path first, the resulting symlinks for the linked packages are broken.
> 
> Quick fix for anyone affected could be to identify symlinks upwards of yarn's link folder and remove them. Proper fix should be in this PR: #7028

My `~/.config` was a symlink created by my `dotfiles` script...
When I ran link with an explicit link-folder that _wasn't_ a symlink or a child of a symlink, it worked!

`yarn link --link-folder ~/.links`
then
`yarn link --link-folder ~/.links @my/package` 🎉

**Comment by aizatto** [comment]
@nemoDreamer thanks for the solution, my `~/.config/` was also a symlink.

I just directly setup:

```
yarn link --link-folder ~/src/dotfiles/config/yarn/link/
```

**Comment by Izhaki** [comment]
Can't get `yarn link` to work with github actions. Need this so my tests and the build see the 'bin' cli tool.

- [with npm link](https://github.com/Izhaki/pliz/commit/7389324d7056cd9e48c17fba1e231daad468097c/checks?check_suite_id=309958497): Works.
- [with yarn link](https://github.com/Izhaki/pliz/commit/991c8448b41f4e44f56ede8d829053618e2504c0/checks?check_suite_id=309982902): Fails.
- [with yarn link --link-folder](https://github.com/Izhaki/pliz/commit/ce1312d6f3d417d1e646554a268ca27b91a57fd7/checks?check_suite_id=309985147): Fails.

**Comment by mfc3370** [comment]
Assuming you have a project in the folder "project-a" and would like to reference it from another "project-b", the following worked for me:

`cd ~/project-a`
`yarn link`

You will get a message
```
yarn link vx.xx.x
success Registered "project-a".
```

Then  do the following...
`cd ~/project-b`
`yarn link project-a`
`yarn add project-a`

After which you can `require('project-a')` or `import from 'project-a'`.

**Comment by Agos88** [comment]
> 
> 
> Pretty sure there’s still something messed up with this command. Here’s what I’m running into:
> 
> ```
> $ cd ~/code/some-module
> 
> $ yarn link
> 
> yarn link v0.17.8
> success Registered "some-module".
> info You can now run `yarn link "some-module"` in the projects where you want to use this module and it will be used instead.
> ✨  Done in 0.07s.
> 
> $ cd ../main-project
> 
> $ yarn link "some-module"
> 
> yarn link v0.17.8
> error No registered module found called "some-module".
> info Visit https://yarnpkg.com/en/docs/cli/link for documentation about this command.
> ```
> 
> What am I missing?
> 
> ![](https://camo.githubusercontent.com/5d40b3bf9197e6eba0b5c04822c15057c704cb3c/687474703a2f2f676966732e6a6f656c676c6f766965722e636f6d2f636f6e74656d70742f6661726c65792d7765656b656e642d7570646174652e676966)

Try to consider the package.json project name

**Comment by cb1kenobi** [comment]
I just ran into the same issues as https://github.com/yarnpkg/yarn/issues/1297#issuecomment-456576816 with Yarn 1.22.4 on macOS. `yarn link` does create the symlink, but the target is a really long relative path that for some reason both the operating system (via `ls`) and Node.js think the symlink is broken.

It's not a permission issue. Maybe there are too many parent directory references and something gets confused? FWIW, I did poke around the Yarn source and found where the symlink was being created, but it wasn't clear how the target was being resolved such that it had so many parent directory references. Perhaps passing a `path.resolve()` around the target when calling `fs.symlink()` would solve it?

As a super hacky workaround, I wrote some code that walks the Yarn links directory and if the file/symlink does not exist (e.g. `fs.existsSync()` returns `false`), but it's a symlink (e.g. `fs.lstatSync().isSymbolicLink()` returns `true`), then I unlink the borked symlink and create a new symlink pointing to `path.resolve(fs.readlinkSync())`.

**Comment by anwarhamr** [comment]
Stuggled w/ this for a few minutes.  Missed the double quotes for the package name.  Seems really unnecessary to need them in general...

**Comment by as-ajitsingh** [comment]
For me following worked-
 In **module-1** - 
```
$ yarn link
```
then in **module-2** -
```
$ yarn link "module-1"
$ yarn add module-1@link:1.0.0
```
Last command will add dependency to package.json of module-2, with a link keyword telling that it is a linked module. After doing that if you do 
```
$ yarn
```
 it will be able to resolve the dependency.

**Comment by jtomaszewski** [comment]
In my case I thought it's not working, but then realized it was actually working, but it's just not working when my app is run inside of a docker container. (Docker containers do not include files from symlinks that lead to files outside of the docker container directory.) More details here: https://github.com/npm/npm/issues/14325 .

Might be helpful to some folks.

**Comment by nfantone** [comment]
Behaviour seems off still - even after all these years. In my experience, sometimes it works, sometimes it just doesn't. The only reliable way I can get linking working is by setting an explicit path in `package.json` as dependency.

So, instead of:

```sh
cd ../some-module
yarn link
success Registered "some-module".
info You can now run `yarn link "some-module"` in the projects where you want to use this package and it will be used instead.
✨  Done in 0.05s.

cd main-app
yarn link "some-module" # Does absolutely nothing
```

I would set:

```json
 "dependencies": {
    "some-module": "../some-module/",
    // ...
  }
```

and then:

```sh
rm yarn.lock
yarn
```

**Comment by monokrome** [comment]
This is still an issue 😢

**Comment by joegasewicz** [comment]
Is anyone working on this issue?

**Comment by stephenhurleyjpl** [comment]
@bestander What's going on with this issue? There's PR #7028 ready to go with minimal changes and a lot of people are waiting on it. You mentioned that linking will be revamped soon, but that was 2017 and it's now 2021. Can we just get the PR approved and fix this bug?

**Comment by austinrivas** [comment]
The yarn official documentation should link to this ticket so that people know not to waste their time trying to link packages like they have with npm for years.

**Comment by hunter2009** [comment]
This is still an issue ! Has any plan to fix it?

**Comment by Karl-EdwardFPJeanMehu** [comment]
It seems as if this may never be worked on....'16 to '21 c'mon!

**Comment by StoneYaoTreelab** [comment]
Creating the `link` with `sudo` works for me.

**Comment by MarsianMan** [comment]
For me (with Yarn 2)  the link seems to work, but unlink does not.

```shell
$ yarn unlink ../../module
Usage Error: Couldn't find a script named "unlink".
```

**Comment by brandongit2** [comment]
@MarsianMan My feeling is that they removed the `unlink` command, but forgot to remove it from the docs. Not sure about the reason for either of those changes though 🤔

**Comment by merceyz** [comment]
@MarsianMan The `yarn unlink` command was added in `3.0.0-rc.3`, unfortunately the docs reflect master and not the latest stable release.

**Comment by qmolab** [comment]
WHY HAS THIS NOT BEEN FIXED YET???

**Comment by jlapier** [comment]
I was trying to use yarn link (yarn 3.0.2) to link to a local folder and was getting an error about workspaces:

```
# from another package, e.g. ~/devel/something-api
❯ yarn link ../something-model
Usage Error: The nearest package directory (/Users/jlapier/devel/something-model) doesn't seem to be part of the project declared in /Users/jlapier.

- If the project directory is right, it might be that you forgot to list devel/something-model as a workspace.
- If it isn't, it's likely because you have a yarn.lock or package.json file there, confusing the project root detection.

$ yarn link [-A,--all] [-p,--private] [-r,--relative] <destination>
```

This was working for another dev on my team so I suspected it was a global config problem. Those who used `sudo` successfully were probably just skipping their global config in their user directory (by acting as a different user, root).

For me, what worked was to clear out some user config, specifically I deleted this file and this directory:

```
~/.yarnrc
~/.config/yarn/link
```

(I'm sure these were created through all the trial and error of trying to get this to work and reading various blog posts over the years of how this _used_ to work.)

**Comment by herleraja** [comment]
I had to downgrade my yarn to 1.22.4 to make it work. 

To check yarn version:
`yarn -v`

To set specific yarn version:
`yarn set version 1.22.4`

**Comment by AgostinhoMayoral** [comment]
rm -rf node_modules (project-a)
rm -rf yarn.lock (project-a)

rm -rf node_modules (project-b)
rm -rf yarn.lock (project-b)

yarn cache clean -f
yarn (project-a)
yarn (project-b)

yarn link (project-b)
rollup --config config/cjs.js -m --watch (project-b)

node ./scripts/linkPackages.js --link project-b --origin  (project-a)

**Comment by p-sebastian** [comment]
For anyone struggling `yarn 3` on a private package

on your package
```bash
yarn link
```

Then on the repo you want the linked package
```bash
# p is for private package 
yarn link -p ~/path/to/my/package
```
This will add this line to the `resolutions` on the `package.json`
```json
"package": "portal:/absolute/path/to/my/package"
```

**Comment by yousseftarekkh** [comment]
@p-sebastian but did that install it in your project and you were able to import it and moreover use it?

**Comment by steinybot** [comment]
`yarn link` is totally unfit for the purpose of testing dependent packages before publishing to a repository. Even if you do get `yarn link` to work (it does for me) `yarn install` will try and resolve it again if it is listed in the `package.json` with a specific version, even if the versions match.

This is such a common use case that every other package manager solves. In the JVM world there has been local publishing of snapshots for several decades.

You may as well give up on `yarn link` and go through the headache of messing up your `package.json` and use 
`yarn add link:/path/to/local/folder` instead.

**Comment by atomoc** [comment]
> Outstanding this does not work even in v4
> 
> I'm running `yarn link @form-atoms/field` in package `@form-atoms/flowbite`.
> 
> Usage Error: Invalid destination '@form-atoms/field'; Can't link the project to itself

similarly

**Comment by justtal** [comment]
> > Outstanding this does not work even in v4
> > I'm running `yarn link @form-atoms/field` in package `@form-atoms/flowbite`.
> > Usage Error: Invalid destination '@form-atoms/field'; Can't link the project to itself
> 
> similarly

Same here... :\

**Comment by devneill** [comment]
Try this for Yarn 4 - it worked for me
https://github.com/yarnpkg/yarn/issues/6880#issuecomment-1910155113

**Comment by Poolshark** [comment]
For everyone using Yarn @3.6.3 (Yarn 3) and workspaces. This worked for me:

Assume the following folder structure:
```
project/
  package1/
    package.json
    ...
  package2/
    package.json
    ...
 package.json
```

1. Link your repo you want to link `yarn link` 
2. Navigate to the project you want to use your local package (eg. `cd ../../projects/project`)
3. Navigate to the package inside your workspace (eg. `cd package1`)
4. Type `yarn workspace package1 link -p <relative-path-to-your-local-package>`
5. Check the `package.json` in the workspace → you should see
```
"resolutions": {
    "your-package-name": "portal:/Users/floragossnig/Code/myPackages/react-hook-useAgent"
  }
```
6. Add the package to the workspace dependencies (or dev-dependencies)
```
"devDependencies": {
  "your-package-name": "*",
  ...
}
```

This worked for me!

**Comment by yojeek** [comment]
Same issue.

```
~/workspace/posenet/burns-audio-wam
❯ yarn link @webaudiomodules/sdk-parammgr
Usage Error: Invalid destination; Can't link the project to itself
❯ yarn -v
3.1.1
```

**Comment by callaginn** [comment]
@Poolshark Running `yarn link -p` works, but it adds a path to our local user folders in package.json. This may be an issue for a lot of people who are sharing package.json across multiple machines.

Is there a way to do that without adding links to your home directory in package.json?

**Comment by Poolshark** [comment]
@callaginn I think I do not quite understand. Why would you share the target repository with a linked local repo? Since the repo is local, you would need to share the local repo source too.

I'm not 100% sure about your use-case but normally the only reason for `yarn link` is to test a package locally before deploying it to NPM.

In any other case I would make the package part of the `yarn workspace` and thus it will be available to everyone who has access to the repository.

**Comment by pepijn-vanvlaanderen** [comment]
Just use `npm link`, it is compatible with yarn and works out of the box.
